platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/testapp/maps/ImageMissingTest.kt

```
@@ -0,0 +1,87 @@
+package com.mapbox.mapboxsdk.testapp.maps
+
+import android.support.test.rule.ActivityTestRule
+import android.support.test.runner.AndroidJUnit4
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.testapp.R
+import com.mapbox.mapboxsdk.testapp.activity.espresso.EspressoTestActivity
+import junit.framework.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@RunWith(AndroidJUnit4::class)
+class ImageMissingTest {
+
+  @Rule
+  @JvmField
+  var rule = ActivityTestRule(EspressoTestActivity::class.java)
+
+  private lateinit var mapView: MapView
+  private val latch = CountDownLatch(1)
+
+  @Test
+  fun testMissingImage() {
+    rule.runOnUiThread {
+      initMap().addOnStyleImageMissingListener {
+        assertEquals("missing-icon", it)
+        latch.countDown()
+      }
+    }
+    latch.await(5, TimeUnit.SECONDS)
```

This should throw an exception if interrupted.